test/integration/preemption_toleration_test.go

```diff
@@ -0,0 +1,307 @@
+				scheduler.WithFrameworkOutOfTreeRegistry(fwkruntime.Registry{preemptiontoleration.Name: preemptiontoleration.New}),
+				scheduler.WithPodInitialBackoffSeconds(int64(0)),
+				scheduler.WithPodMaxBackoffSeconds(int64(0)),
+			)
+			defer testutils.CleanupTest(t, testCtx)
+
+			cs, ns := testCtx.ClientSet, testCtx.NS.Name
+
+			// Create test node
+			if _, err := cs.CoreV1().Nodes().Create(testCtx.Ctx, node, metav1.CreateOptions{}); err != nil {
+				t.Fatalf("failed to create node: %v", err)
+			}
+
+			// create test priorityclass for evaluating PreemptionToleration policy
+			if _, err := cs.SchedulingV1().PriorityClasses().Create(testCtx.Ctx, tt.priorityClass, metav1.CreateOptions{}); err != nil {
+				t.Fatalf("failed to create priorityclass %s: %v", tt.priorityClass.Name, err)
+			}
+
+			// Create victim candidate pod and it should be scheduled
+			if _, err := cs.CoreV1().Pods(ns).Create(testCtx.Ctx, victimCandidate, metav1.CreateOptions{}); err != nil {
+				t.Fatalf("failed to create victim candidate pod %q: %v", victimCandidate.Name, err)
+			}
+			if err := wait.Poll(1*time.Second, 60*time.Second, func() (bool, error) {
+				return podScheduled(cs, ns, victimCandidate.Name), nil
+			}); err != nil {
+				t.Fatalf("victim candidate pod %q failed to be scheduled: %v", victimCandidate.Name, err)
+			}
+			// simulate pod scheduled time
+			if err := wait.Poll(1*time.Second, 60*time.Second, func() (bool, error) {
+				vc, err := cs.CoreV1().Pods(ns).Get(testCtx.Ctx, victimCandidate.Name, metav1.GetOptions{})
+				if err != nil {
+					return false, err
+				}
+				vc.Status.Conditions = []v1.PodCondition{{
+					Type:               v1.PodScheduled,
+					Status:             v1.ConditionTrue,
+					LastTransitionTime: metav1.Time{Time: time.Now()},
+				}}
+				_, err = cs.CoreV1().Pods(ns).UpdateStatus(testCtx.Ctx, vc, metav1.UpdateOptions{})
+				if err != nil {
+					return false, err
+				}
+				return true, nil
+			}); err != nil {
+				t.Fatalf("failed to update victim candidate pod %q scheduled time: %v", victimCandidate.Name, err)
+			}
+
+			// Create the preemptor pod.
+			<-time.After(tt.waitPreemptorCreation)
+			if _, err := cs.CoreV1().Pods(ns).Create(testCtx.Ctx, tt.preemptor, metav1.CreateOptions{}); err != nil {
+				t.Fatalf("failed to create preemptor Pod %q: %v", tt.preemptor.Name, err)
+			}
+
+			if tt.canTolerate {
+				// - the victim candidate pod keeps being scheduled, and
+				// - the preemptor pod is not scheduled
+				if err := consistently(1*time.Second, 15*time.Second, func() (bool, error) {
+					a := podScheduled(cs, ns, victimCandidate.Name)
+					b := podScheduled(cs, ns, tt.preemptor.Name)
+					t.Logf("%s scheduled=%v, %s scheduled = %v", victimCandidate.Name, a, tt.preemptor.Name, b)
+					return a && !b, nil
+				}); err != nil {
+					t.Fatalf("preemptor pod %q was scheduled: %v", tt.preemptor.Name, err)
+				}
+			} else {
+				// - the preemtor pod got scheduled successfully
```

Comment:
preemptor

Reply:
fixed in https://github.com/kubernetes-sigs/scheduler-plugins/pull/224/commits/1df617784e33151cba48e94bfb88c498205a9cb3